File: extensions/aether-market/kotlin/src/main/kotlin/market/PoVService.kt
package market

import java.util.UUID

interface IPoVService {
    suspend fun issueToken(token: PoVToken): PoVToken
    suspend fun revokeToken(tokenId: UUID, reason: String): Boolean
    suspend fun getScore(subjectId: UUID): PoVScore
    suspend fun getTokensFor(subjectId: UUID): List<PoVToken>
    suspend fun getTokensBy(issuerId: UUID): List<PoVToken>
    suspend fun verifyToken(tokenId: UUID): Boolean
    suspend fun syncTokens(peerNodeId: String): Int
}

class PoVServiceImpl : IPoVService {

    override suspend fun issueToken(token: PoVToken): PoVToken {
        TODO("not implemented")
    }

    override suspend fun revokeToken(tokenId: UUID, reason: String): Boolean {
        TODO("not implemented")
    }

    override suspend fun getScore(subjectId: UUID): PoVScore {
        TODO("not implemented")
    }

    override suspend fun getTokensFor(subjectId: UUID): List<PoVToken> {
        TODO("not implemented")
    }

    override suspend fun getTokensBy(issuerId: UUID): List<PoVToken> {
        TODO("not implemented")
    }

    override suspend fun verifyToken(tokenId: UUID): Boolean {
        TODO("not implemented")
    }

    override suspend fun syncTokens(peerNodeId: String): Int {
        TODO("not implemented")
    }
}

File: extensions/aether-market/swift/Sources/AetherNetMarket/PoVService.swift
import Foundation

public protocol PoVServiceProtocol {
    func issueToken(_ token: PoVToken) async throws -> PoVToken
    func revokeToken(tokenId: UUID, reason: String) async throws -> Bool
    func getScore(subjectId: UUID) async throws -> PoVScore
    func getTokensFor(subjectId: UUID) async throws -> [PoVToken]
    func getTokensBy(issuerId: UUID) async throws -> [PoVToken]
    func verifyToken(tokenId: UUID) async throws -> Bool
    func syncTokens(peerNodeId: String) async throws -> Int
}

public class PoVService: PoVServiceProtocol {

    public init() {}

    public func issueToken(_ token: PoVToken) async throws -> PoVToken {
        fatalError("not implemented")
    }

    public func revokeToken(tokenId: UUID, reason: String) async throws -> Bool {
        fatalError("not implemented")
    }

    public func getScore(subjectId: UUID) async throws -> PoVScore {
        fatalError("not implemented")
    }

    public func getTokensFor(subjectId: UUID) async throws -> [PoVToken] {
        fatalError("not implemented")
    }

    public func getTokensBy(issuerId: UUID) async throws -> [PoVToken] {
        fatalError("not implemented")
    }

    public func verifyToken(tokenId: UUID) async throws -> Bool {
        fatalError("not implemented")
    }

    public func syncTokens(peerNodeId: String) async throws -> Int {
        fatalError("not implemented")
    }
}
